PacketDotNet/IPv6ExtensionHeader.cs

```diff
@@ -0,0 +1,89 @@
+﻿using PacketDotNet.MiscUtil.Conversion;
+using PacketDotNet.Utils;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+/*
+This file is part of PacketDotNet
+
+PacketDotNet is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+PacketDotNet is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with PacketDotNet.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/*
+ * Copyright 2018 Steven Haufe<haufes@hotmail.com>
+ */
+
+namespace PacketDotNet
+{
+    [Serializable]
+    public class IPv6ExtensionHeader
```

Comment:
I think this should be more like the TcpOptions, so you would have different kinds of IPv6ExtensionHeader, such as RoutingHeader, MobilityHeader, etc.

Reply:
(the rest of this looks good by the way)

Reply:
I'll look at breaking this out a bit more, but I'm not going to get into fully parsing out the different headers.

Reply:
I understand, it doesn't need to happen right now, if this is ever required in the future it can be added then. For now this base would work mostly fine.